File: GestionProjet/src/main/java/com/example/gestionprojet/services/interfaces/AiAssistantService.java
package com.example.gestionprojet.services.interfaces;

import com.example.gestionprojet.dto.AssigneeSuggestionRequest;
import com.example.gestionprojet.dto.AssigneeSuggestionResponse;
import com.example.gestionprojet.dto.DiscussionSummaryResponse;
import com.example.gestionprojet.dto.RiskTasksResponse;
import com.example.gestionprojet.dto.TaskDescriptionRequest;
import com.example.gestionprojet.dto.TaskDescriptionResponse;

public interface AiAssistantService {

    /** Genere une description riche + criteres + estimation a partir d'un titre. */
    TaskDescriptionResponse describeTask(TaskDescriptionRequest request);

    /** Propose les meilleurs membres pour prendre une tache. */
    AssigneeSuggestionResponse suggestAssignee(AssigneeSuggestionRequest request);

    /** Identifie les taches d'un projet a risque de glissement. */
    RiskTasksResponse detectRiskTasks(Long projectId);

    /** Resume une longue discussion (commentaires) d'un projet. */
    DiscussionSummaryResponse summarizeDiscussion(Long projectId);
}

File: gestion-projet-frontend/src/app/core/services/ai-assistant.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import {
  AssigneeSuggestionRequest,
  AssigneeSuggestionResponse,
  DiscussionSummaryResponse,
  RiskTasksResponse,
  TaskDescriptionRequest,
  TaskDescriptionResponse
} from 'src/app/shared/models/ai-assistant.model';

@Injectable({
  providedIn: 'root'
})
export class AiAssistantService {
  private readonly base = '/api/ai';

  constructor(private http: HttpClient) {}

  /** Genere description + criteres + estimation a partir d'un titre court. */
  describeTask(req: TaskDescriptionRequest): Observable<TaskDescriptionResponse> {
    return this.http.post<TaskDescriptionResponse>(`${this.base}/task/describe`, req);
  }

  /** Suggere les meilleurs membres pour prendre une tache. */
  suggestAssignee(req: AssigneeSuggestionRequest): Observable<AssigneeSuggestionResponse> {
    return this.http.post<AssigneeSuggestionResponse>(`${this.base}/task/suggest-assignee`, req);
  }

  /** Identifie les taches a risque dans un projet. */
  detectRiskTasks(projectId: number): Observable<RiskTasksResponse> {
    return this.http.get<RiskTasksResponse>(`${this.base}/project/${projectId}/risk-tasks`);
  }

  /** Resume la discussion d'un projet (jusqu'a 60 commentaires les + recents). */
  summarizeDiscussion(projectId: number): Observable<DiscussionSummaryResponse> {
    return this.http.get<DiscussionSummaryResponse>(`${this.base}/project/${projectId}/discussion-summary`);
  }
}
